components/brave_wallet_ui/panel/container.tsx

```diff
@@ -166,7 +166,9 @@ function Container (props: Props) {
     if (selectedPanel === 'connectWithSite') {
       const foundDefaultAccountInfo = defaultAccounts.find(account => connectingAccounts.includes(account.address.toLowerCase()))
       const foundDefaultAccount = accounts.find((account) => account.address.toLowerCase() === foundDefaultAccountInfo?.address?.toLowerCase() ?? '')
-      if (foundDefaultAccount) {
+      // Checking selectedAccounts length here to ensure that
+      // we only update this once on mount.
+      if (foundDefaultAccount && selectedAccounts.length === 0) {
```

Comment:
let's check selectedAccounts.length === 0 first, that way we don't need to loop over anything if default is already set

Reply:
Updated!